### PR TITLE
NOTICES Move timeouts from actions to lifecycle methods

### PR DESCRIPTION
**Pull request by artpi**
Timeout for notice duration is now moved from redux actions to component lifecycle methods.

Plays much better while dehydrating store.
CC @gwwar

### CODE REVIEW

File: client/components/notice/index.jsx [hunk 1]
```diff
@@ -12,9 +12,11 @@ import Gridicon from 'components/gridicon';
 
 export default React.createClass( {
 	displayName: 'Notice',
+	dismissTimeout: null,
 
 	getDefaultProps() {
 		return {
+			duration: null,
```
Comment (gwwar) [inline client/components/notice/index.jsx]: don't forget to add this to prop types

File: client/components/notice/index.jsx [hunk 2]
```diff
@@ -35,6 +37,18 @@ export default React.createClass( {
 		className: PropTypes.string
 	},
 
+	componentDidMount() {
+		if ( this.props.duration > 0 ) {
+			this.dismissTimeout = setTimeout( this.props.onDismissClick, this.props.duration );
+		}
+	},
+
+	componentWillMount() {
```
Comment (gwwar) [inline client/components/notice/index.jsx]: Do you mean `componentWillUnmount` ?

Comment (artpi) [inline client/components/notice/index.jsx]: Yeah, fixed :)

File: client/components/notice/index.jsx (hunk 1 above)
Comment (artpi) [inline client/components/notice/index.jsx]: Done!

**Comment by gwwar** [comment]
:ship:

**Pull request merged**